src/Chapter3.hs

```diff
@@ -971,6 +1092,42 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Weekday = Monday | Tuesday | Wednesday | Thursday | Friday | Saturday | Sunday
+  deriving (Show, Eq, Bounded, Enum)
+
+isWeekend :: Weekday -> Bool
+isWeekend day
+  | day == pred maxBound  = True
+  | day == maxBound       = True
+  | otherwise             = False
```

Comment:
This is a nice and generic implementation!
Though, you can improve it even further to make it work automatically even if you decide to change to the American style of weekdays